### ISSUE TITLE
Help for the point layer and its base map using Lambert Conic for Maps of Canada

### ISSUE THREAD

**Issue by Mengrou** [GeoDaCenter/geoda#1797]
Dear all,

    My name is Mengrou and I am a beginner in GIS software. It is really exciting to explore the GeoDa! Recently, I am working on a project. I need to use GeoDa and the coordinates list in Canada (.csv) that I have to generate maps. Now my coordinates list looks like the following:

Postal Codes   latitude    longitude
   A1B2C3          47.57        -52.68

   And the obs are almost 890,000.

   Since I found that StatsCan use the Lambert Conic projection, I need to convert the geographical coordinates to Lambert Conic in STATA. My thought is after the conversion, I would have two new variables for Lambert Conic and after saving it as .csv, I would import it into GeoDa and generate a point layer based on this projection. After conversion,  now it looks like

Postal Codes latitude    longitude       lon        lat
   A1B2C3         47.57      -52.68           0.435   1.15

   Now it has more than 890,000 obs, with some obs have lambert conic but no postal codes and with some obs have neither of them. 
   I then saved this new file as csv and imported it into GeoDa using the lat and lon in the csv file. Compared to the maps from StatsCan, the point layer seems to have similar outline but still looks different. Another question is that, I used both the map from StatsCan and the point layer that I generated to find their base maps. GeoDa didn't tell me that there is no proper projection for this point layer but compared to the base map of the StatsCan, this base map also seems to be weird.  I also attach them for the reference. If anyone knows why there are such kind of problems and can let me know, I will be very grateful!

Thank you so much!

Best,

Mengrou
   
![base map of maps from statscan](https://user-images.githubusercontent.com/46332209/50985783-2a30b180-14d3-11e9-882a-09dea85e81cb.png)
![base map of point layer](https://user-images.githubusercontent.com/46332209/50985784-2ac94800-14d3-11e9-9ef4-e32c222653db.png)
![maps after converting the projection](https://user-images.githubusercontent.com/46332209/50985785-2ac94800-14d3-11e9-949f-b38353dd48f4.png)
![maps from statscan](https://user-images.githubusercontent.com/46332209/50985786-2ac94800-14d3-11e9-9bb8-cf33a2e61580.png)

**Comment by lixun910** [comment]
@Mengrou It looks like a projection issue to me. What are the projections used in the points dataset (2nd and 3rd figures)? and does the 4th figure work with a basemap in GeoDa?  

GeoDa automatically detects/recognizes most of the projection information if it is available when load the data into GeoDa. If you can share the data with me, I can take a look at it. 

Thanks!

**Comment by Mengrou** [comment]
Hi Professor Li,

    Many thanks for your reply! 

    Yes, I think there may have some problems with the projection. For the points dataset, I used the Lambert Conformal Conic as the projection. My point dataset has the postal codes and coordinates for Canada, but since I want to use the maps from StatsCan as my base map, which I downloaded from "https://www12.statcan.gc.ca/census-recensement/alternative_alternatif.cfm?l=eng&dispext=zip&teng=lcar000b16a_e.zip&k=%20%20%20%2030304&loc=http://www12.statcan.gc.ca/census-recensement/2011/geo/bound-limit/files-fichiers/2016/lcar000b16a_e.zip", I had to convert the coordinates into Lambert Conformal Conic. This is the projection that StatsCan uses. I did this step in STATA. I then import the dataset which has the projection to GeoDa and produced the 2nd figure. And the 3rd figure is the map that I try to fit in the base map. 

     I got the figure 4 from the StatsCan zip file and found that it really works with GeoDa. That makes me feel that maybe I made some mistakes in projection. Prof, I am wondering if I can convert the projection in GeoDa and make the StatsCan as the base map? And for the dataset that I am using, since there may have some confidential data, I would ask my supervisor tomorrow and see if I can share it with you. So I may not send it to you now. But many thanks for your help!

Best,

Mengrou

**Comment by Mengrou** [comment]
Dear Professor Li,

    Many thanks for your reply! 

    Yes, I think there may have some problems with the projection.

     For the points dataset, I used the Lambert Conformal Conic as the projection. My point dataset has the postal codes and coordinates for Canada, but since I want to use the maps from StatsCan as my base map, which I downloaded from "https://www12.statcan.gc.ca/census-recensement/alternative_alternatif.cfm?l=eng&dispext=zip&teng=lcar000b16a_e.zip&k=%20%20%20%2030304&loc=http://www12.statcan.gc.ca/census-recensement/2011/geo/bound-limit/files-fichiers/2016/lcar000b16a_e.zip", I had to convert the coordinates into Lambert Conformal Conic. This is the projection that StatsCan uses. I did this step in STATA. I then import the dataset which has the projection to GeoDa and produced the 2nd figure. And the 3rd figure is the map that I try to fit in the base map. 

     I got the figure 4 from the StatsCan zip file and found that it really works with GeoDa. That makes me feel that maybe I made some mistakes in projection. Prof, I am wondering if I can convert the projection in GeoDa and make the StatsCan as the base map? And for the dataset that I am using, since there may have some confidential data, I would ask my supervisor tomorrow and see if I can share it with you. So I may not send it to you now. But many thanks for your help!

Best,

Mengrou

**Comment by Mengrou** [comment]
Thanks for your advice, prof! I haven't done that but I would try it! I am just wondering how can I open the png file you sent to me because it seems that when I tried to click on it, it always goes to the top of this page. 

Thank you again!

**Comment by lixun910** [comment]
Here is the screenshot, use the 3rd button (from left) to add a new layer (points data set) after load StatsCan data in GeoDa. If GeoDa recognizes the projection of the points dataset, it should be converted to the projection StatsCan used automatically. 

<img width="265" alt="screen shot 2019-01-10 at 1 50 06 pm" src="https://user-images.githubusercontent.com/2423887/51002294-1182c480-14f0-11e9-9468-1bff01d07d6c.png">

**Comment by lixun910** [comment]
@Mengrou any updates?

**Comment by Mengrou** [comment]
Dear Professor Li,

I am trying to use the method you suggested, but I find that the GeoDa version that I use (GeoDa1.12) doesn't have the button of adding a new layer nor does it have the fourth button. Is it because of the version? Thank you so much!

Best,

Mengrou

**Comment by lixun910** [comment]
@Mengrou Can you try this version: Windows -- https://www.dropbox.com/s/4jb4bafjwv9g9f9/GeoDa-1.12.1.185-Windows-64bit.exe?dl=0

**Comment by Mengrou** [comment]
Hi Prof,

Thanks for sending me the link. Now I can use the button to add layers. Actually, I haven't used the weird point layer that I created yesterday to try the button. I am now just using three good maps from StatsCan to learn the function of this button. The maps' outlines can be overlapped perfectly and the only difference is the level of detail. 

 If my understanding is correct, as the images show, the map that was first imported to GeoDa would always be the current map and it is colorful. Since it's always the current map, the corresponding table is also always the info for this map. Other maps' data would not be shown in the table and their colors are just black-white. (As is shown in the current map image) However, as the layers' order changes in "Map Layer Settings", the map's color and level of detail would change and how it changes is up to the map that is on top of the others and that is open.  (As is shown in the maps after adding layers image).

But Prof, I have a question about this button. I'm not quite clear about the function of this button because I find that every time when I click on somewhere of a map, it would only show the linked information in the current map's table. So I cannot figure out the difference between adding layers and not adding them. 

Thank you so much, prof! Thanks for your help! 

![current map image](https://user-images.githubusercontent.com/46332209/51059048-d7c9c080-15b8-11e9-89b0-65f976a337e0.png)
![maps after adding layers](https://user-images.githubusercontent.com/46332209/51059049-d8625700-15b8-11e9-9747-7f82de7ae215.png)



Best,

Mengrou

**Comment by lixun910** [comment]
@Mengrou you are welcome! Yes, you are right that GeoDa only has one working layer, which is the map you loaded first time. All other layers loaded by the "+" button are for references or visualization. There are some spatial functions you can apply, e.g. Table->Spatial Join, you can apply spatial join by using two layers.  That's why the table is always for the first loaded map. So, if you work with one map, please remember to load it into GeoDa first.

To change the color of the loaded layers, you can right click the layer name in the "Map Layer Settings", and from the pop-up menu, you will find menu options to change the color etc. 

I noticed you have two more layers loaded, and they are all polygon layers so they overlay on each other. However, the multi-layer function in GeoDa is more for the case that you have one polygon layer and multiple points layers, so you can do spatial count (e.g. how many points in each polygon, or which polygon is associated with which point), and apply association between points and polygons for visualization purpose.

Let me know if the points datasets work as multi-layers in GeoDa.

**Comment by lixun910** [comment]
@Mengrou When you do select operation on multi-layer maps, it only applies on the top layer. So, you will not see highlighted items in table if the top layer is not associated with the "first" layer. 

In the right click pop-up menu in "May layer settings", you can find menu options "Set Highlight Association". Using this tool, you can setup the rules how to trigger other layers to highlight when you select on top layer.

 For example, you have a top layer with provinces (has a id column), and a second layer with counties and each county has a field "province_id" showing which province this county is associated with. Then, you can "Set Highlight Association" for county layer and field "province_id" is associated to "id" field in top layer. By doing this, you will see highlighting triggered by this rule in county Table (if you load county map first in GeoDa)

**Comment by Mengrou** [comment]
Thank you so much for your explanation, Prof! 

Currently, I have the map from StatsCan whose dbs file only has postal codes and province id. What I want to do in GeoDa is to show the distribution of Canadian dairy producers' info. 

If my understanding is correct, first, I have to import the dataset of Canadian dairy producers' info. I also should generate coordinates using (maybe) other softwares based on address info. And I must convert the coordinates to the projection that the base map uses. (Actually, I try to add the point layer that I generated to the map from StatsCan but it still doesn't work. I think it's the problem of projection just as what you told me before and it is a very important problem. Without solving this problem, I cannot head to the next steps).
 
Then, I should use the dataset with coordinates and projection to make a point layer and click on the "base map" button. Prof, here, I have two questions. First, is it necessary for me to use the map from StatsCan as the base map rather than the original world map in GeoDa? This is because I found that the original GeoDa world map can be zoomed but it contains too many places.  Second, if it is necessary, is it possible for me to use the StatsCan map as the base map?

Thank you so much, Prof! And I am wondering whether Mac also has this version since adding layers is a very important tool for my work. Thanks for your time! 

Best,

Mengrou
